src/Tools/dotnet-monitor/Auth/MonitorApiKeyConfigurationObserver.cs

```diff
@@ -49,6 +51,11 @@ private void OnMonitorApiKeyOptionsChanged(MonitorApiKeyConfiguration options)
 
         private void CheckMonitorApiKeyOptions(MonitorApiKeyConfiguration options)
         {
+            if (_authConfigurationOptions.KeyAuthenticationMode == KeyAuthenticationMode.NoAuth)
```

Comment:
I think you can move this to the Initialize method so that way it doesn't even have to listen to the changed at all.